### PR TITLE
BGV Rotate Kernel 

### PR DESCRIPTION
**Pull request by christopherngutierrez**
## Proposed changes

Adds Rotate Kernel for BGV. Tested for 16k, 32k, 64k and 128k Poly mod. 
rotate.py - implementation of rotate kernel
manifest.json - adds rotate entry to BGV manifest

## Types of changes

What types of changes does your code introduce to the HE Toolkit project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you are unsure about any of them, do not hesitate to ask. We are
here to help! This is simply a reminder of what we are going to look for before
merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/IntelLabs/hec-p-isa-tools/blob/main/CONTRIBUTING.md) agreement
- [x] Current formatting and unit tests / base functionality passes locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: kerngen/pisa_generators/rotate.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+"""Module containing implementation for rotate kernel."""
+
+from dataclasses import dataclass
+
+from high_parser.pisa_operations import PIsaOp, Comment
+from high_parser import KernelContext, HighOp, Polys, KeyPolys
+
+from .basic import Add, mixed_to_pisa_ops
+from .relin import KeyMul, DigitDecompExtend, extract_last_part_polys
```
Comment (faberga) [inline kerngen/pisa_generators/rotate.py:12]:
```suggestion
from .basic import Add, KeyMul, mixed_to_pisa_ops, extract_last_part_polys
from .decomp import DigitDecompExtend
```

Comment (faberga) [reply kerngen/pisa_generators/rotate.py:12]: Hi @christopherngutierrez, remember that KeyMul, extract_last_part_polys, and DigitDecompExtend were moved out from relin.py by the refactoring introduced by PR #47

**Review by faberga — CHANGES_REQUESTED** [review]
See suggested code changes in rotate.py

**Review by faberga — APPROVED** [review]

**Review by faberga — CHANGES_REQUESTED** [review]
Hi @christopherngutierrez 
Now that we have the bug fix for the NTT, let's re-run the tests for 16K, 32K, 64K and 128K polynomials.

**Comment by christopherngutierrez** [comment]
> Hi @christopherngutierrez Now that we have the bug fix for the NTT, let's re-run the tests for 16K, 32K, 64K and 128K polynomials.

@faberga confirmed working for NTT/INTT, Relin, and Rotate kernels for 16k, 32k, 64K, and 128K.

**Review by faberga — APPROVED** [review]

**Review by faberga — APPROVED** [review]

**Review by jobottle — APPROVED** [review]
Looks good

**Pull request merged**